src/raft/tests.rs

```diff
@@ -940,12 +1055,99 @@ async fn snapshot_install_unreliable_crash_2d() {
     snap_common(false, false, true).await;
 }
 
+/// do the servers persist the snapshots, and restart using snapshot along
+/// with the tail of the log?
+#[madsim::test]
+async fn snapshot_all_crash_2d() {
+    let servers = 3;
+    let iters = 5;
+
+    let t = RaftTester::new_with_snapshot(servers).await;
+    info!("Test (2D): crash and restart all servers");
+    let mut random = rand::rng();
+
+    t.one(random.gen_entry(), servers, true).await;
+
+    for _iter in 0..iters {
+        // enough to get a snapshot
+        for _ in 0..=SNAPSHOT_INTERVAL {
```

Comment:
the original go version uses this:
`	nn := (SnapShotInterval / 2) + (rand.Int() % SnapShotInterval `